webapp/tests/forms/steps/test_steuerlotse_step.py

```diff
@@ -471,61 +544,117 @@ def test_return_an_instance_of_the_class_input_form(self):
         assert type(created_form) == form_step.InputForm
         assert dir(created_form) == dir(form_step.InputForm()) # check that it has the correct input fields
 
-@pytest.mark.usefixtures("test_request_context")
-class TestFormSteuerlotseStepUpdateData:
 
-    def test_if_form_valid_then_updates_empty_data_correctly(self):
-        req_form_data = {'date': ['12', '12', '1980'], 'decimal': '42', 'pet': 'lizard'}
-        validated_data = MockFormWithInputStep.validate_data(ImmutableMultiDict(req_form_data), req_form_data)
+class TestFormSteuerlotseStepPrepareRenderInfo:
+
+    def test_stored_data_set_and_form_data_empty_then_create_form_with_stored_data(self):
+        input_form_data = ImmutableMultiDict({})
+        stored_data = {'date': datetime.date(1980, 12, 12), 'decimal': Decimal('42'), 'pet': 'lizard'}
+
+        render_info = MockFormWithInputStep.prepare_render_info(stored_data=stored_data, input_data=input_form_data)
+
+        assert render_info.form.data == stored_data
+
+    def test_form_data_set_and_stored_data_empty_then_create_form_with_form_validated_data(self):
+        input_form_data = ImmutableMultiDict({'date': ['1', '3', '2019'], 'decimal': '42', 'pet': 'lizard'})
+        stored_data = {}
+
+        render_info = MockFormWithInputStep.prepare_render_info(stored_data=stored_data, input_data=input_form_data)
+
+        assert render_info.form.data == {'date': datetime.date(2019, 3, 1), 'decimal': Decimal('42'), 'pet': 'lizard'}
+
+    def test_form_data_set_and_stored_data_set_then_create_form_with_combination_from_form_and_stored_data(self):
+        input_form_data = ImmutableMultiDict({'decimal': '42', 'pet': 'lizard'})
+        stored_data = {'date': datetime.date(2019, 3, 1)}
+
+        render_info = MockFormWithInputStep.prepare_render_info(stored_data=stored_data, input_data=input_form_data)
+
+        assert render_info.form.data == {'date': datetime.date(2019, 3, 1), 'decimal': Decimal('42'), 'pet': 'lizard'}
+
+    def test_if_should_update_false_and_form_valid_then_do_update(self):
+        input_form_data = ImmutableMultiDict({'date': ['12', '12', '1980'], 'decimal': '42', 'pet': 'lizard'})
+        stored_data = {}
+        should_update_data = False
 
-        updated_data = MockFormWithInputStep.update_data({}, validated_data)
+        render_info = MockFormWithInputStep.prepare_render_info(stored_data=stored_data, input_data=input_form_data, should_update_data=should_update_data)
 
-        assert updated_data == {'date': datetime.date(1980, 12, 12), 'decimal': 42.00, 'pet': 'lizard'}
+        assert render_info.stored_data == {}
```

Comment:
You mean `test_if_should_update_false_and_form_valid_then_do_not_update` (not)

Reply:
And can you move this down to the other tests for updating data? e.g. `test_if_form_invalid_and_should_update_then_do_not_call_update`

Reply:
Yes, thank you